# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/plan9port/cross.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- plan9port-d0d4408/INSTALL.vanilla	2021-10-19 18:08:45.000000000 +0200
+++ plan9port-d0d4408/INSTALL	2022-01-08 11:40:25.741000000 +0100
@@ -99,6 +99,11 @@
 	rm -f ./a.out
 fi
 
+if [ -z "$HOSTPLAN9" ]; then
+	# Native compiling, set HOSTPLAN9 to PLAN9
+	HOSTPLAN9="$PLAN9"
+fi
+
 if [ `uname` != Darwin ]; then
 	# Determine whether fontsrv X11 files are available.
 	rm -f a.out
@@ -127,12 +132,17 @@
 
 cd src
 if $dobuild; then
-	echo "* Building mk..."
-	../dist/buildmk 2>&1 | sed 's/^[+] //'
+	if [ -e "$HOSTPLAN9/bin/mk" ]; then
+		# Cross-compiling, Use host make
+		PATH=$PATH:$HOSTPLAN9/bin
+	else
+		echo "* Building mk..."
+		../dist/buildmk 2>&1 | sed 's/^[+] //'
 
-	if [ ! -x ../bin/mk ]; then
-		echo "* Error: mk failed to build."
-		exit 1
+		if [ ! -x ../bin/mk ]; then
+			echo "* Error: mk failed to build."
+			exit 1
+		fi
 	fi
 
 	echo "* Building everything (be patient)..."
